### ISSUE TITLE
remove `cargo` dependency

### ISSUE THREAD

**Issue by MarcoIeni** [MarcoIeni/release-plz#1331]
## Motivations

cargo increases the compile time of release-plz.
We only use it to download crates from the registries.

## Solution

After reading [this](https://doc.rust-lang.org/cargo/reference/registry-index.html) docs, I think we can implement the download of the crate by ourselves.

Example code doing this (not sure if they work with private registries):
- https://github.com/Nullus157/cargo-dl
- https://github.com/Xion/cargo-download
- https://github.com/nakabonne/rhack

Things to check:
- [ ] make sure it works with private registries: both sparse and git protocol
- [ ] support authentication with cargo login. This is from `cargo help login`:

      For the default cargo:token credential provider, the token is saved in $CARGO_HOME/credentials.toml.
      CARGO_HOME defaults to .cargo in your home directory.

      If a registry has a credential-provider specified, it will be used.
      Otherwise, the providers from the config value registry.global-credential-providers will be attempted, starting from the end of the list.
  

## Alternatives

It would be much better if cargo offered this out of the box. I.e. `cargo download my-crate`.
We could open an RFC to add this functionality to cargo.
However, this might take a lot of time until we see it implemented.

**Comment by mbodmer** [comment]
I am not sure whether you would get all the information you need from `cargo search <CRATENAME> [--registry myregistry]`.

Do we need to actually download the crate?

**Comment by MarcoIeni** [comment]
Yes. Release-plz compares the current project with the published version file by file, going back to history commit by commit.

When the two projects match release-plz stops looking back in git history and that is the last commit that goes into the changelog for the new release.

**Comment by MarcoIeni** [comment]
The best would be to have this command built into cargo, i.e. https://github.com/rust-lang/cargo/issues/1861
